vm/devices/storage/disk_nvme/nvme_driver/src/namespace.rs

```diff
@@ -499,6 +504,52 @@ impl Namespace {
 
         Ok(())
     }
+
+    /// Return Namespace ID.
+    pub fn nsid(&self) -> u32 {
+        self.nsid
+    }
+
+    /// Save namespace data for servicing.
+    pub fn save(&self) -> anyhow::Result<SavedNamespaceData> {
+        let id = self.state.identify.lock();
+        let mut save_data = SavedNamespaceData {
+            nsid: self.nsid,
+            block_count: self.state.block_count.load(Ordering::Relaxed),
+            identify_ns: [0; 4096],
+        };
+        save_data.identify_ns.copy_from_slice(id.as_bytes());
+        Ok(save_data)
+    }
+
+    /// Restore namespace data after servicing.
+    pub(super) fn restore(
+        driver: &VmTaskDriver,
+        admin: Arc<Issuer>,
+        rescan_event: Arc<event_listener::Event>,
+        identify_ctrl: Arc<spec::IdentifyController>,
+        io_issuers: &Arc<IoIssuers>,
+        device_id: &str,
+        nsid: u32,
+        identify_ns: &[u8; 4096],
+        _saved_state: &SavedNamespaceData,
+    ) -> Result<Self, NamespaceError> {
+        let identify = nvm::IdentifyNamespace::read_from_prefix(identify_ns)
+            .unwrap_or(nvm::IdentifyNamespace::new_zeroed());
+        // Restore provides Identify Namespace result to new() so there is no wait.
```

Comment:
We should add an explicit `restore()` that is a non-async function, rather than have this `block_on` here.

Reply:
Thanks. I was thinking to add an explicit Namespace::restore but wanted to avoid duplicate code. If this is preferred way then I will redesign Namespace::new to use common logic with restore.

One remaining question for me is creation of async rescan task when restoring namespace. I may need help with that.

Reply:
Done